test/e2e/nodepool_kv_advanced_multinet_test.go

```diff
@@ -0,0 +1,208 @@
+	}
+	infraClient, err := cm.DiscoverKubevirtClusterClient(k.ctx, k.client, k.hostedCluster.Spec.InfraID, creds, localInfraNS, np.GetNamespace())
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	vmis := &kubevirtv1.VirtualMachineInstanceList{}
+	labelSelector := labels.SelectorFromValidatedSet(labels.Set{hyperv1.NodePoolNameLabel: np.Name})
+	g.Eventually(func(gg Gomega) {
+		gg.Expect(
+			infraClient.GetInfraClient().List(k.ctx, vmis, &crclient.ListOptions{Namespace: guestNamespace, LabelSelector: labelSelector}),
+		).To(Succeed())
+
+		gg.Expect(vmis.Items).To(HaveLen(1))
+		vmi := vmis.Items[0]
+		// Use gomega HaveField so we can skip "Mac" matching
+		matchingInterface := &kubevirtv1.Interface{}
+		gg.Expect(vmi.Spec.Domain.Devices.Interfaces).To(ContainElement(
+			HaveField("Name", "iface0_"+k.infraNamespace()+"-net1"), matchingInterface),
+		)
+		gg.Expect(matchingInterface.InterfaceBindingMethod.Bridge).ToNot(BeNil())
+		gg.Expect(vmi.Spec.Networks).To(ContainElement(kubevirtv1.Network{
+			Name: "iface0_" + k.infraNamespace() + "-net1",
+			NetworkSource: kubevirtv1.NetworkSource{
+				Multus: &kubevirtv1.MultusNetwork{
+					NetworkName: k.infraNamespace() + "/net1",
+				},
+			},
+		}))
+	}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
+}
+
+func (k KubeVirtAdvancedMultinetTest) BuildNodePoolManifest(defaultNodepool hyperv1.NodePool) (*hyperv1.NodePool, error) {
+
+	nadYAML := fmt.Sprintf(`
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  namespace: %[1]s
+  name: %[2]s
+spec:
+  config: |2
+    {
+            "cniVersion": "0.3.1",
+            "name": "l2-network",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"layer2",
+            "netAttachDefName": "%[1]s/%[2]s"
+    }
+`, k.infraNamespace(), "net1")
+	nad := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(nadYAML), &nad); err != nil {
+		return nil, fmt.Errorf("failed unmarshaling net-attach-def: %w", err)
+	}
+	if err := k.client.Create(context.Background(), nad); err != nil {
+		return nil, fmt.Errorf("failed creating net-attach-def: %w", err)
+	}
+
+	// Make infra namespace privileged
+	annotatedHostedCluster := k.hostedCluster.DeepCopy()
+	annotatedHostedCluster.Annotations[hyperv1.PodSecurityAdmissionLabelOverrideAnnotation] = "privileged"
+	if err := k.client.Patch(context.Background(), annotatedHostedCluster, client.MergeFrom(k.hostedCluster)); err != nil {
+		return nil, fmt.Errorf("failed updating hosted cluster to have privileged infra namespace: %w", err)
+	}
+
+	dnsmasqPod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: k.infraNamespace(),
+			Name:      "dnsmasq",
+			Annotations: map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(`[{"interface":"net1","name":"net1","namespace":"%s"}]`, k.infraNamespace()),
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
```

Comment:
this Pod might need to be pinned to a specific node. The issue is that we're making the infra master nodes scheduable in CI. The control plane infra nodes (or master nodes) are not metal instances, so VMs cannot schedule there. but this dnsmasq pod could technically.

it might make sense to pick a node and pin both the dnsmasq and VM to it. take a look at the `test/e2e/nodepool_kv_nodeselector_test.go` for an example of how that might be accomplished.

Reply:
Make sense to move it out of control plane, but there is no need to have it next to VM since we are using a L2 overlay between metal nodes and it works as expected.